shared/web_features_manifest_github_download_test.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2024 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+//go:build small
+
+package shared
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-github/v47/github"
+	"github.com/migueleliasweb/go-github-mock/src/mock"
```

Comment:
Is this a widely used library that has ongoing contributions in the future? My concern is that e.g. if GitHub API upgrades to v48 with breaking changes, the mock might break

Reply:
Alternatively, we can generate our [own mock(https://github.com/web-platform-tests/wpt.fyi/blob/main/shared/sharedtest/metadata_util_mock.go) through declaring an interface in MockGen

Reply:
That is always a valid concern. I picked this library because it was recommended by the GitHub API client we use.

https://github.com/google/go-github?tab=readme-ov-file#testing-code-that-uses-go-github

It allows you see the URL path to check exactly what the github client sends off

But we could create a small interface for the GetLatestRelease method. I can do that.